### ISSUE TITLE
Feature: Install addon by name instead of URL

### ISSUE THREAD

**Issue by vvambo** [Trojan295/eso-addons#42]
It should not be too complicated to take a keyword and use it with the esoui search.  
The search on the website already delivers the addon page if only one result fits, if there are multiple, the names could be parsed and given into a selection (1-X) with the option to deny and check through the search results on esoui

```
Found <x> Addons searching for "<keyword>":
<URLtoESOUISearch> # clickable link 
1. <name-searchresult1>
2. <name-searchresult2>
3. ...

Select the result by typing the according number, 'n' to abort or insert the URL for an addon manually.
```

Kind of what I'd imagine

**Comment by Trojan295** [comment]
Good idea, I'm just a bit concerned that this will base on web scrapping of the esoui.com webpage and in case they change the page layout this could break.

On the other hand, current functionalities (like getting the addon download link) also scrap the HTML of esoui.com.

@vvambo, do you want to contribute such a feature?

**Comment by vvambo** [comment]
I have not seen esoui.com change in ~5 years? So I think the risk for that is small.

I'd love to help, but I have 0 experience with either Rust or web APIs.
I can try to dig through the  esoui wiki/docs  though.

**Comment by mahoromax** [comment]
Just thought about this again, two other ideas:

-  IF skipping the web scrapping, it could output the search URL (https://www.esoui.com/downloads/search.php?search=<keyword> ) in case the search didnt directly go to an Addon page (= search.php is still part of the URL)
And then the user opens that, finds the correct one and pastes the URL

**Comment by vvambo** [comment]
Just thought about this again, two other ideas:

IF skipping the web scrapping, it could output the search URL (https://www.esoui.com/downloads/search.php?search= ) in case the search didnt directly go to an Addon page (= search.php is still part of the URL)
And then the user opens that, finds the correct one and pastes the URL

**Comment by mhutter** [comment]
In case there was still any doubt, scraping their website is explicitly prohibited in [their Terms of Service](https://mmoui.com/?tos), section 4: 

> You agree not to access (or attempt to access) any of the Services by any means other than through the interface that is provided by GGMODS. You specifically agree not to access (or attempt to access) any of the Services through any automated means (including use of scripts or web crawlers).

---

What _would_ be an useful feature would be to open the search page directly, e.g. using the [`open` crate](https://crates.io/crates/open)
